clients/wallet/src/stellar_wallet.rs

```diff
@@ -80,14 +166,69 @@ impl StellarWallet {
 		Ok(transactions)
 	}
 
+	/// Submits transactions found in the wallet's cache to Stellar.
+	/// Returns a tuple: (list of txs successfully submitted, list of errors of txs not resubmitted)
+	pub async fn resubmit_transactions_from_cache(
+		&mut self,
+	) -> (Vec<TransactionResponse>, Vec<Error>) {
+		let _ = self.transaction_submission_lock.lock().await;
+		let horizon_client = Client::new();
+
+		let mut passed = vec![];
+
+		let (envs, mut errors) = match self.cache.get_tx_envelopes() {
+			Ok(x) => x,
+			Err(errors) => return (passed, errors),
+		};
+
+		for env in envs.into_iter() {
+			match self.submit_transaction(env, &horizon_client).await {
+				Ok(response) => passed.push(response),
+				Err(e) => errors.push(e),
```

Comment:
@b-yap 
are we able to add logs if `submit_transaction` failed? 
For now there is no logging and the value returned from `resubmit_transactions_from_cache` not handling at all.
`let _ = wallet.resubmit_transactions_from_cache().await;`  (line 441)